api/src/wfl/module/covid.clj

```diff
@@ -266,6 +259,53 @@
                     (->> (jdbc/insert! tx terra-executor-table)))]
     [terra-executor-type (-> items first :id str)]))
 
+(defn ^:private import-snapshot!
+  "Return snapshot reference for ID imported to WORKSPACE as NAME."
+  [{:keys [workspace] :as _executor}
+   {:keys [name id]   :as _snapshot}]
+  (rawls/create-snapshot-reference workspace id name))
+
+(defn ^:private from-source
+  "Coerce SOURCE-ITEM to form understood by EXECUTOR via FROMSOURCE."
+  [{:keys [fromSource] :as executor}
+   source-item]
+  (cond (= "importSnapshot" fromSource) (import-snapshot! executor source-item)
+        :else (throw (ex-info "Unknown fromSource" {:executor executor}))))
+
+(defn ^:private create-submission!
+  "TO IMPLEMENT:
+  Create submission from REFERENCE."
+  [executor reference])
+
+(defn ^:private write-workflows!
+  "Write WORKFLOWS to DETAILS table."
+  [{:keys [details]                :as _executor}
+   {:keys [referenceId]            :as _reference}
+   {:keys [submissionId workflows] :as _submission}]
+  (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+    (let [now      (OffsetDateTime/now)
+          id-start (inc (postgres/table-max tx details :id))
```

Comment:
Want to flag how I've generated ids here, since we could be writing multiple rows (one submission can trigger many workflows).